src/display/text_layer.js

```diff
@@ -540,55 +561,53 @@ var renderTextLayer = (function renderTextLayerClosure() {
         this._expanded = true;
         this._bounds.length = 0;
       }
-      if (expandDivs) {
-        for (var i = 0, ii = this._textDivs.length; i < ii; i++) {
-          var div = this._textDivs[i];
-          var transform;
-          var width = div.dataset.originalWidth;
-          if (div.dataset.canvasWidth !== undefined && width > 0) {
-            // Dataset values are of type string.
-            var textScale = div.dataset.canvasWidth / width;
+
+      for (var i = 0, ii = this._textDivs.length; i < ii; i++) {
+        var div = this._textDivs[i];
+        var divProperties = this._textDivProperties.get(div);
+
+        if (expandDivs) {
+          var transform = '';
+          var width = divProperties.originalWidth;
+          var textScale = 1;
+
+          if (divProperties.canvasWidth !== 0 && width > 0) {
+            textScale = divProperties.canvasWidth / width;
             transform = 'scaleX(' + textScale + ')';
-          } else {
-            transform = '';
           }
-          var rotation = div.dataset.angle;
-          if (rotation) {
+          var rotation = divProperties.angle;
```

Comment:
Nit: Again, you might just get rid of this variable and use `divProperties.angle` directly below (similar to the `divProperties.padding...` ones below).